crates/sui-core/src/authority_aggregator.rs

```diff
@@ -201,9 +218,71 @@ impl<A> AuthorityAggregator<A> {
             metrics,
             timeouts,
             safe_client_metrics,
+            network_client_metrics,
+            committee_store,
         }
     }
 
+    /// This function recreates AuthorityAggregator with the given committee.
+    /// It also updates committee store which impacts other of its references.
+    /// If it is called on a Validator/Fullnode, it **may** interleave with the the authority active's
+    /// reconfiguration process, and leave the commmittee store in an inconsistent state.
+    /// When catching up to the latest epoch, it should call `reconfig_from_genesis` first to fill in
+    /// all previous epoch's committee info.
+    pub fn recreate_with_net_addresses(
+        &self,
+        committee: CommitteeWithNetAddresses,
+        network_config: &Config,
+    ) -> SuiResult<AuthorityAggregator<NetworkAuthorityClient>> {
+        let network_clients = make_network_authority_client_sets_from_committee(
+            &committee,
+            network_config,
+            self.network_client_metrics.clone(),
+        )
+        .map_err(|err| SuiError::GenericAuthorityError {
+            error: format!("Failed to make authority clients from committee: {:?}", err),
+        })?;
+
+        let safe_clients = network_clients
+            .into_iter()
+            .map(|(name, api)| {
+                (
+                    name,
+                    SafeClient::new(
+                        api,
+                        self.committee_store.clone(),
+                        name,
+                        self.safe_client_metrics.clone(),
+                    ),
+                )
+            })
+            .collect::<BTreeMap<_, _>>();
+
+        // TODO: It's likely safer to do the following operations atomically, in case this function
+        // gets called from different threads. It cannot happen today, but worth the caution.
+        let new_committee = committee.committee;
+        fp_ensure!(
+            self.committee.epoch + 1 == new_committee.epoch,
```

Comment:
why is is not ok to create a new instance of AA which is for a further epoch in the future but only the next one from this instance?

Reply:
the authority aggregator (more specifically, its safe clients) needs to know about every epoch's committee information, in order to perform sanity checks. This check makes sure we fail fast when there's missing committee information